rs/tests/driver/src/driver/bootstrap.rs

```diff
@@ -567,13 +518,111 @@ fn node_to_config(node: &Node) -> NodeConfiguration {
     }
 }
 
-fn configure_setupos_image(
+#[derive(Clone)]
+pub enum NestedVersionTarget {
+    Mainnet,
+    /// When true, use the branch -test variant.
+    Branch(bool),
+    Published {
+        version: String,
+        url: Url,
+        sha256: String,
+    },
+}
```

Comment:
I wonder if it wouldn't be better to specify which version to use for the SetupOS via environment variables instead of via Rust code. Similar to what we're proposing in: https://github.com/dfinity/ic/pull/675. 

One of the advantages is that you can retarget a system-test by just changing a single line in a BUILD.bazel without changing and rebuilding Rust code. Another advantage is having multiple versions of the same test that each target a different SetupOS.